addons/blocklydevtools/userscript.js

```diff
@@ -0,0 +1,419 @@
+
+        container.setAttribute("data-isblocklydeveditor", "true"); //Give it an identifying attribute
+        container.contentEditable = true; //Allow it's contents to be edited
+
+        //Styling
+        container.style.padding = "10px";
+        container.style.userSelect = "text";
+        container.style.cursor = "auto";
+        container.style.font = "12pt monospace";
+        container.style.boxShadow = "0px 0px 10px rgba(0,0,0,0.5)";
+        container.style.color = "yellowgreen";
+        container.style.border = "2px solid white";
+        container.style.zIndex = "998";
+        container.style.borderRadius = "0.8rem";
+        container.style.width = "max-content";
+        container.style.backgroundColor = "rgb(0,0,30)";
+        container.style.overflowX = "hidden";
+        container.style.maxHeight = "100vh";
+        container.style.overflowY = "scroll";
+        container.style.caretColor = "white";
+
+        //Disable spellcheck and autocomplete.
+        container.setAttribute("autocomplete", "false");
+        container.setAttribute("spellcheck", "false");
+
+        //Set the editor's contents to the formatted original XML.
+        container.innerHTML = formatXml(originalXml, true);
+
+        function update() { //Function called when the workspace updates
+            originalXml = getXmlFromBlockId(blockId, dom); //Update original XML
+            if (!originalXml) { //If it is now undefined, the block has been deleted or no longer exists, so remove listeners and exit.
+                workspace.removeChangeListener(update);
+                return;
+            }
+            container.innerHTML = formatXml(originalXml, true); //Update contents
+        }
+
+        workspace.addChangeListener(update); //Register change listener
+
+        container.save = function () { //Function to save
+            var xmlStr = (Blockly.Xml.domToText || Blockly.utils.xml.domToText)(dom); //Convert the dom to string
+            if (!xmlStr.includes(originalXml)) { //If the domn does not contain the original block XML, it is impossible to save changes.
+                throw new Error("Workspace XML does not contain block!");
+            }
+            xmlStr = xmlStr.replace(originalXml, container.textContent); //Replace original XML with modified XML.
+            //xmlStr = xmlStr.replace(/\u2003/g, "").replace(/\n/g, "");
+            xmlStr = xmlStr.replace(/\u00A0/g, "\u0020"); //Replace any non-breaking spaces with normal ones.
+            dom = (Blockly.Xml.textToDom || Blockly.utils.xml.textToDom)(xmlStr); //Update the DOM variable.
+            workspace.clear(); //Clear the workspace
+            Blockly.Xml.domToWorkspace(dom, workspace); //Load the DOM
+            if (Blockly.getMainWorkspace().getToolbox() //If the blockly instance has a toolbox, it needs to be refreshed,
+            ) {
+                Blockly.getMainWorkspace().getToolbox().refreshSelection();
```

Comment:
Can't the existing `workspace` here?
```suggestion
            if (workspace.getToolbox() //If the blockly instance has a toolbox, it needs to be refreshed,
            ) {
                workspace.getToolbox().refreshSelection();
```

Reply:
Done.